src/overlays/ovl_i5/fox_ti.c

```diff
@@ -1,138 +1,5375 @@
+                D_i5_801BBEF0[16] = 0;
+                D_i5_801BBEF0[17] = 0;
+                boss->swork[0x17] = 0;
+                D_i5_801BBEF0[6] = 0x50;
+                D_i5_801BBEF0[7] = 0;
+                for (i = 0; i < ARRAY_COUNTU(D_i5_801BBF00); i++) {
+                    if (!(D_i5_801BBF00[i].unk_26 & 4)) {
+                        D_i5_801BBF00[i].unk_26 = 3;
+                    }
+                }
+            }
+            break;
+    }
+
+    switch (boss->swork[34]) {
+        case 0:
+            break;
+        case 1:
+            boss->fwork[48] += boss->fwork[2];
+            if (Animation_GetFrameCount(&D_70096EC) * 2 <= boss->fwork[48]) {
+                boss->fwork[48] -= Animation_GetFrameCount(&D_70096EC) * 2;
+            } else if (boss->fwork[48] < 0.0f) {
+                boss->fwork[48] += Animation_GetFrameCount(&D_70096EC) * 2 - 1.0f;
+            }
+            break;
+        case 2:
+            boss->fwork[48] += 1.0f;
+            if (Animation_GetFrameCount(&D_A009990) * 2 <= boss->fwork[48]) {
+                if (boss->state == 0xD) {
+                    boss->fwork[0] = 0.0f;
+                    boss->swork[34] = 1;
+                    boss->fwork[48] = 0;
+                } else {
+                    boss->fwork[48] = 0;
+                }
+            }
+            break;
+    }
+    if (spB8 == 1) {
+        boss->unk_04C = 0;
+        boss->fwork[0] = 0.0f;
+        boss->swork[3]--;
+        if (boss->swork[3] <= 0) {
+            boss->swork[1] = (s32) (Rand_ZeroOne() * 6.0f) + 7;
+            func_i5_80193B30(boss);
+        }
+        boss->swork[4]++;
+        if (boss->swork[37] & 0x18) {
+            D_i5_801BBEF0[44]++;
+        }
+        D_i5_801BBEF0[42] = 0;
+
+        switch (boss->swork[1]) {
+            case 7:
+                if (boss->swork[37] & 2) {
+                    // not necessarily j
+                    j = 0;
+                    for (i = 0; i < 4; i++) {
+                        if (boss->swork[i + 9] <= 0) {
+                            j++;
+                        }
+                    }
+
+                    if ((j != 4) && gTeamShields[3] > 0) {
+                        Radio_PlayMessage(gMsg_ID_4095, RCID_PEPPY);
+                    }
+                    boss->swork[37] &= ~2;
+                }
+                break;
+            case 9:
+                if (boss->swork[37] & 1) {
+                    if ((boss->swork[29] != 0) && gTeamShields[1] > 0) {
+                        Radio_PlayMessage(gMsg_ID_4099, RCID_FALCO);
+                    }
+                    boss->swork[37] &= ~1;
+                }
+                break;
+        }
+    }
+
+    while (boss->fwork[0x31] >= 360.0f) {
+        boss->fwork[0x31] -= 360.0f;
+    }
+
+    while (boss->fwork[0x31] < 0.0f) {
+        boss->fwork[0x31] += 360.0f;
+    }
+
+    for (i = 0; i < 4; i++) {
+        if (boss->swork[i + 9] == 0) {
+            j = boss->swork[i + 0x11];
```

Comment:
```suggestion
            j = boss->swork[17 + i];
```
when indexing work buffers like this, it's better to have the offset first.